File: src/androidTest/java/com/owncloud/android/ui/activity/UserInfoActivityIT.java
/*
 * Nextcloud Android client application
 *
 * @author Andy Scherzinger
 * Copyright (C) 2020 Andy Scherzinger
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package com.owncloud.android.ui.activity;

import android.content.Intent;

import com.owncloud.android.AbstractIT;
import com.owncloud.android.lib.common.UserInfo;
import com.owncloud.android.utils.ScreenshotTest;

import org.junit.Rule;
import org.junit.Test;
import org.parceler.Parcels;

import androidx.test.espresso.intent.rule.IntentsTestRule;

public class UserInfoActivityIT extends AbstractIT {
    @Rule
    public IntentsTestRule<UserInfoActivity> activityRule = new IntentsTestRule<>(UserInfoActivity.class,
                                                                                  true,
                                                                                  false);

    @Test
    @ScreenshotTest
    public void fullUserInfoDetail() {
        final Intent intent = new Intent(targetContext, UserInfoActivity.class);
        intent.putExtra(UserInfoActivity.KEY_ACCOUNT, user);
        UserInfo userInfo = new UserInfo("test",
                                         true,
                                         "Firstname Familyname",
                                         "oss@rocks.com",
                                         "+49 7613 672 255",
                                         "Awesome Place Av.",
                                         "https://www.nextcloud.com",
                                         "nextclouders",
                                         null,
                                         null
        );
        intent.putExtra(UserInfoActivity.KEY_USER_DATA, Parcels.wrap(userInfo));
        UserInfoActivity sut = activityRule.launchActivity(intent);

        shortSleep();
        shortSleep();

        screenshot(sut);
    }
}

File: app/src/androidTest/java/com/owncloud/android/ui/activity/UserInfoActivityIT.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2025 Alper Ozturk <alper.ozturk@nextcloud.com>
 * SPDX-FileCopyrightText: 2020 Andy Scherzinger <info@andy-scherzinger.de>
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.ui.activity

import android.content.Intent
import androidx.test.core.app.launchActivity
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.isRoot
import com.owncloud.android.AbstractIT
import com.owncloud.android.lib.common.UserInfo
import com.owncloud.android.utils.ScreenshotTest
import org.junit.Test

class UserInfoActivityIT : AbstractIT() {
    private val testClassName = "com.owncloud.android.ui.activity.UserInfoActivityIT"

    @Test
    @ScreenshotTest
    fun fullUserInfoDetail() {
        val intent = Intent(targetContext, UserInfoActivity::class.java).apply {
            putExtra(UserInfoActivity.KEY_ACCOUNT, user)

            val userInfo = UserInfo(
                "test",
                true,
                "Firstname Familyname",
                "oss@rocks.com",
                "+49 7613 672 255",
                "Awesome Place Av.",
                "https://www.nextcloud.com",
                "nextclouders",
                null,
                null
            )
            putExtra(UserInfoActivity.KEY_USER_DATA, userInfo)
        }

        launchActivity<UserInfoActivity>(intent).use { scenario ->
            val screenShotName = createName(testClassName + "_" + "fullUserInfoDetail", "")
            onView(isRoot()).check(matches(isDisplayed()))

            scenario.onActivity { sut ->
                screenshotViaName(sut, screenShotName)
            }
        }
    }
}
